File: bench/trees.go
package main
import "fmt"
type T struct{ l, r *T }
func mk(d int) *T { if d == 0 { return &T{} }; return &T{mk(d - 1), mk(d - 1)} }
func (t *T) check() int { if t.l == nil { return 1 }; return 1 + t.l.check() + t.r.check() }
func main() {
  fmt.Println(mk(21).check()); acc := 0
  for d := 4; d <= 20; d += 2 { n := 16 * (20 + 4 - d); for i := 0; i < n; i++ { acc += mk(d).check() } }
  fmt.Println(acc) }

File: bench/trees.rs
struct T { l: Option<Box<T>>, r: Option<Box<T>> }
fn mk(d: i32) -> Box<T> { if d == 0 { Box::new(T { l: None, r: None }) } else { Box::new(T { l: Some(mk(d - 1)), r: Some(mk(d - 1)) }) } }
fn check(t: &T) -> i64 { match (&t.l, &t.r) { (Some(l), Some(r)) => 1 + check(l) + check(r), _ => 1 } }
fn main() { println!("{}", check(&mk(21))); let mut acc = 0i64;
  let mut d = 4; while d <= 20 { let n = 16 * (20 + 4 - d); for _ in 0..n { acc += check(&mk(d)); } d += 2; } println!("{}", acc); }
